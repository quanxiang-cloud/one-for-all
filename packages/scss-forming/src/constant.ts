import sorting from 'postcss-plugin-sorting';
import removeComments from 'postcss-discard-comments';
import mergeSelector from 'postcss-combine-duplicated-selectors';
import { parse, stringify } from 'postcss-scss';

export const processOptions = {
  from: 'index.scss',
  to: 'index.css',
  parser: parse,
  stringifier: stringify,
  map: false,
};

const sortOptions = {
  order: [
    'custom-properties',
    'dollar-variables',
    'at-variables',
    {
      type: 'atrule',
      name: 'include',
    },
    'declarations',
    {
      type: 'rule',
      selector: /^&:\w+$/
    },
    'rules',
    'atrule'
  ],
  'properties-order': 'alphabetical',
};

export const plugins = [removeComments, sorting(sortOptions), mergeSelector];
