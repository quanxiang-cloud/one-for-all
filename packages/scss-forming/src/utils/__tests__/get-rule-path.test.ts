import fs from 'fs';
import path from 'path';
import postcss from 'postcss';
import { plugins, processOptions } from '../../constant';
import getRulePath from '../get-rule-path';

test('rulePath', async () => {
  const scss = fs.readFileSync(path.join(__dirname, './fixtures/index.scss'), { encoding: 'utf-8' });
  const { root } = postcss(plugins).process(scss, processOptions);

  const ruleSet = new Set<string>();
  root.walkRules((rule) => {
    const path = getRulePath(rule);
    if (path) {
      ruleSet.add(path.join('/'));
    }
  });

  expect(Array.from(ruleSet).sort()).toMatchSnapshot();
});
