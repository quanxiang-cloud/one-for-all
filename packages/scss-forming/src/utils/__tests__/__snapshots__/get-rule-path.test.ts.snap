// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rulePath 1`] = `
Array [
  ".foo",
  ".foo/.bar",
  ".foo/.bar/.abc",
  ".foo/.bar/.abc/.def",
  ".foo/.bar/.abc/.xyz",
  ".foo/.bar/.baz",
  ".foo/.bar/.baz:first-child",
]
`;
